File: src/benchmarks/pink/benchmark/Benchmark.java
package pink.benchmark;

public class Benchmark {

  public static class Phasor {

    private double cur_phase = 0.0;
    double[] buffer = new double[64];
    double freq, phase;
    double phase_incr;
    int buffer_size;

    public Phasor(double freq, double phase, int sr, int buffer_size) {
       this.freq = freq;
       this.phase = phase;
       this.phase_incr = freq / sr;
       this.buffer_size = buffer_size;
    }

    public double[] tick() {
      double phs = this.phase;

      for(int i = 0; i < buffer_size; i++) {
        buffer[i] = phs;
        phs = (phs + phase_incr) / 1.0;
      }

      this.phase = phs;

      return buffer;
    }

  }

}

File: src/benchmarks/pink/benchmark/benchmark.clj
(ns pink.benchmark.benchmark
  (:import [pink.benchmark Benchmark$Phasor])
  (:require [pink.config :refer :all]
            [pink.oscillators :refer :all]
            [pink.util :refer :all]
            [clojure.pprint :refer [pprint]]
            ))

(defn run-phasor-test
  []
  (let [p (Benchmark$Phasor. 440.0 0.0 *sr* *buffer-size*)]
    (println "Java Phasor Test")
    (doseq [_ (range 5)] 
      (time 
        (doseq [_ (range 1000000)]
          (.tick p)))))
  (let [p (phasor 440.0 0.0)]
    (println "Pink Phasor Test")
    (doseq [_ (range 5)] 
      (time 
        (doseq [_ (range 1000000)]
          (p))))))

(defn run-mul-test
  []
  (let [p (mul 440.0 2.0)]
    (println "Pink Mul Test")
    (doseq [_ (range 5)] 
      (time 
        (doseq [_ (range 1000000)]
          (p)))))
  (let [p (mul2 440.0 2.0)]
    (println "Pink Mul2 Test")
    (doseq [_ (range 5)] 
      (time 
        (doseq [_ (range 1000000)]
          (p))))))


(comment 

(run-mul-test)
(run-phasor-test)

)

